SleepNumberController_App.groovy

```diff
@@ -190,82 +248,132 @@ def installed() {
+  if ((Boolean)settings.variableRefresh && (di <= iZ || ni <= iZ)) {
+    logError "Invalid refresh intervals ${di} or ${ni}"
   }
-  if (settings.variableRefreshModes) {
+  if ((Boolean)settings.variableRefreshModes) {
     subscribe(location, "mode", configureVariableRefreshInterval)
   }
-  setRefreshInterval(0 /* force picking from settings */, "" /* ignored */)
+  remTsVal("lastBedInfoUpdDt")
+  remTsVal("lastBedDataUpdDt")
+  remTsVal("lastFamilyDataUpdDt")
+  remTsVal("lastPrivacyDataUpdDt")
+  remTsVal("lastSleepNumDataUpdDt")
+  remTsVal("lastSleeperDataUpdDt")
+  setRefreshInterval(0.0 /* force picking from settings */, "" /* ignored */)
   initializeBedInfo()
   refreshChildDevices()
   updateLabel()
 }
 
 void updateLabel() {
   // Store the user's original label in state.displayName
-  if (!app.label.contains("<span") && state?.displayName != app.label) {
-    state.displayName = app.label
-  }
-  if (state?.status || state?.paused) {
-    def status = state?.status
-    String label = "${state.displayName} <span style=color:"
-    if (state?.paused) {
-      status = "(Paused)"
+  String al= (String)app.label
+  if (!al.contains("<span") && (String)state.displayName != al) {
+    state.displayName = al
+  }
+  if ((String)state.status || (Boolean)state.paused) {
+    String status, nstatus
+    status = (String)state.status
+    nstatus=status
+    String label
+    label = "${state.displayName} <span style=color:"
+    if ((Boolean)state.paused) {
+      nstatus = "(Paused)"
       label += "red"
-    } else if (state.status == "Online") {
+    } else if (status == "Online") {
       label += "green"
-    } else if (state.status.contains("Login")) {
+    } else if (status.contains("Login")) {
       label += "red"
     } else {
       label += "orange"
     }
-    label += ">${status}</span>"
+    label += ">${nstatus}</span>"
     app.updateLabel(label)
   }
 }
 
 void initializeBedInfo() {
-  debug "Setting up bed info"
-  def bedInfo = getBeds()
-  state.bedInfo = [:]
-  bedInfo.beds.each() { Map bed ->
-    debug "Bed id ${bed.bedId}"
-    if (!state.bedInfo.containsKey(bed.bedId)) {
-      state.bedInfo[bed.bedId] = [:]
-    }
-    def components = []
-    for (def component : bed.components) {
-      if (component.type == "Base"
-          && component.model.toLowerCase().contains("integrated")) {
-        // Integrated bases need to be treated separately as they don't appear to have
-        // foundation status endpoints so don't lump this with a base type directly.
-        components << "Integrated Base"
-      } else {
-        components << component.type
+  if(debugOk()) logDebug "Initializing bed info"
+  Map bedInfo = getBeds()
+  if(bedInfo){
+    updTsVal("lastBedInfoUpdDt")
+    Map<String,Map> sbedInfo = [:]
+    state.bedInfo = sbedInfo
+    ((List<Map>)bedInfo.beds).each() { Map bed ->
+      String bdId=bed.bedId.toString()
+      if(debugOk()) logDebug "Bed id ${bdId}"
```

Comment:
given the logDebug method already checks if debugOk is true, is there an advantage to checking here as well?

Reply:
It just avoids creating the string argument repeatedly. (Optimization vs. correctness)